lib/libPlugin.lua

```diff
@@ -0,0 +1,724 @@
+--[[
+  This file is part of darktable,
+  copyright (c) 2016 Bill Ferguson
+  
+  darktable is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  darktable is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+libPlugin = {}
+
+local dt = require "darktable"
+local log = require "lib/libLog"
+
+require "lib/dtutils"
+
+-- put the plugin_manager widgets here to prevent namespace pollution
+
+libPlugin.placeholder = dt.new_widget("separator"){}
+
+-- format options container
+-- TODO: read the value from saved preferences 
+
+libPlugin.jpeg_slider = dt.new_widget("slider"){
+  label = "quality",
+  sensitive = true,
+  soft_min = 5,      -- The soft minimum value for the slider, the slider can't go beyond this point
+  soft_max = 100,     -- The soft maximum value for the slider, the slider can't go beyond this point
+  hard_min = 5,       -- The hard minimum value for the slider, the user can't manually enter a value beyond this point
+  hard_max = 100,    -- The hard maximum value for the slider, the user can't manually enter a value beyond this point
+  value = 90          -- The current value of the slider
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.png_bit_depth = dt.new_widget("combobox"){
+  label = "bit depth",
+  value = 1, 8, 16,
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.tif_bit_depth = dt.new_widget("combobox"){
+  label = "bit depth",
+  value = 1, 8, 16, 32,
+}
+
+-- TODO: read the value from saved preferences 
+
+libPlugin.tif_compression = dt.new_widget("combobox"){
+  label = "compression",
+  value = 1, "uncompressed", "deflate", "deflate with predictor", "deflate with predictor (float)",
+}
+
+libPlugin.tif_widget = dt.new_widget("box"){
+  orientation = "vertical",
+  libPlugin.tif_bit_depth,
+  libPlugin.tif_compression,
+}
+
+-- TODO: read the value from saved preferences 
+-- TODO: add more formats
+
+libPlugin.format_combobox = dt.new_widget("combobox"){
+  label = "file format",
+  value = 1, "JPEG (8-bit)", "PNG (8/16-bit)", "TIFF (8/16/32-bit)",
+  changed_callback = function(self)
+    log.msg(log.debug, "value is " .. self.value)
```

Comment:
yes, but also remember that when someone else is using your lib and debuging his own code it's not good to flood with log messages.even debug messages. Your library can probably use DEBUG to signal "weird" behaviour (typically printing a DEBUG message before raising an exception, which is something the person is probably hunting) but not whenever you change a value.